yorkie/rpc/server.go

```diff
@@ -217,18 +230,31 @@ func (s *Server) DetachDocument(
 	}
 
 	// if pack.HasChanges() {
-	lockKey := fmt.Sprintf("pushpull-%s", pack.DocumentKey.BSONKey())
-	if err := s.backend.MutexMap.Lock(lockKey); err != nil {
+	locker, err := s.backend.MutexMap.NewLocker(
+		ctx,
+		sync.NewKey(fmt.Sprintf("pushpull-%s", pack.DocumentKey.BSONKey())),
```

Comment:
The `fmt.Sprintf("pushpull-%s", pack.DocumentKey.BSONKey())` code seems to be used in several places. How about subtracting with a separate function?

Reply:
Yes, this is a reasonable opinion. However, the abstraction for Key seems to be early now, so how about proceeding later?

Reply:
Yes, as you said, it would be better to proceed later :D Thank you for your opinion.